### PR TITLE
Add sentinel1.grd subpackage

### PR DESCRIPTION
**Pull request by lossyrob**
**Related Issue(s):**
https://github.com/stactools-packages/sentinel1-grd/issues/2

**Description:**
This PR moves the code for processing Sentinel1 GRD that currently exists in https://github.com/stactools-packages/sentinel1-grd into a subpackage of this project.

The project is refactored to include two subpackages - `rtc` and `grd`. The existing code is moved into the `rtc` subpackage and the code from sentinel1-grd is moved to the `grd` subpackage. The top-level `stac sentinel1` command is now a group that has `rtc` and `grd` as subcommands, which behave how the previous `stac sentinel1` and `stac sentinel1grd` commands worked, respectively.

I've updated the README to account for these two datasets along with the examples and tests.

__NOTE:__ The GRD code was created by @maximlamare, all credit to them - I'm just moving it over. @maximlamare if you'd like to rebase these commits and force push to this branch so that your account is associate with the commits by all means do so! 

**PR checklist:**

- [*] Code is formatted (run `scripts/format`).
- [*] Code lints properly (run `scripts/lint`).
- [*] Tests pass (run `scripts/test`).
- [*] Documentation has been updated to reflect changes, if applicable.
- [*] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

**Comment by lossyrob** [comment]
@maximlamare invited you to this repository; I can't assign you as a reviewer until that invite is accepted, but would like you to please review!

**Review by maximlamare — APPROVED** [review]
There were a few details that I forgot to change in the setup.cfg, but the merge has taken care of them. Looks all good to me. I appreciate the help I received with this.

**Comment by scottyhq** [comment]
thanks @lossyrob! I'm going to try a release of the current code on pypi since we never did that, then we can add this PR to the CHANGELOG and make a new release

File: src/stactools/sentinel1/grd/properties.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    Returns:
+        pystac.Asset: An asset with the SAR relevant properties.
+    """
+    # Read meta file
+    root = XmlElement.from_file(href)
+
+    # Fixed properties
+    sar_ext.frequency_band = FrequencyBand("C")
+    sar_ext.center_frequency = 5.405
```
Comment (scottyhq) [inline src/stactools/sentinel1/grd/properties.py:23]: in the future would be good to consolidate common properties for GRD,RTC,SLC, etc under `sentinel1/properties.py`

File: src/stactools/sentinel1/grd/metadata_links.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        granule_href: str,
+    ):
+        self.granule_href = granule_href
+        self.href = os.path.join(granule_href, "manifest.safe")
+
+        root = XmlElement.from_file(self.href)
+        data_object_section = root.find("dataObjectSection")
+        if data_object_section is None:
+            raise ManifestError(
+                f"Manifest at {self.href} does not have a dataObjectSection")
```
Comment (scottyhq) [inline src/stactools/sentinel1/grd/metadata_links.py:27]: @maximlamare just wanted to point out https://github.com/bopen/xarray-sentinel/blob/main/xarray_sentinel/esa_safe.py , which seems to be a nice approach for parsing SAFE files using the accompanying .xsd files for XML validation and parsing

**Review by scottyhq — APPROVED** [review]

File: src/stactools/sentinel1/grd/metadata_links.py (hunk 2 above)
Comment (maximlamare) [reply src/stactools/sentinel1/grd/metadata_links.py:27]: Once we have contributed all sensors we had in mind, I could take a look and propose an update to account for this.

**Pull request merged**